gpg/backend_test.go

```diff
@@ -48,6 +49,72 @@ func TestBackend_CRUDImportedKey(t *testing.T) {
 	testAccStepReadKey(t, b, storage, "test", nil, true)
 }
 
+func TestBackend_Signing(t *testing.T) {
+	b, storage := getTestBackend(t)
+
+	keyData := map[string]interface{}{
+		"real_name":  "Vault",
+		"email":      "vault@example.com",
+		"comment":    "Comment",
+		"key_bits":   4096,
+		"exportable": true,
+	}
+	base64InputData := "bXkgc2VjcmV0IGRhdGEK"
+	otherBase64InputData := "c29tZSBvdGhlciBkYXRhCg=="
+	masterName := "test"
+	testAccStepCreateKey(t, b, storage, masterName, keyData, false)
+
+	t.Run("signing with master key", func(t *testing.T) {
+		signature := testAccStepSign(t, b, storage, masterName, base64InputData)
+		testAccStepVerify(t, b, storage, masterName, map[string]interface{}{
+			"input":     base64InputData,
+			"signature": signature,
+		}, true)
+		testAccStepVerify(t, b, storage, masterName, map[string]interface{}{
```

Comment:
You might also wish to test bad signatures, bad marshaling/hashing algorithms, etc, but we have coverage of that [elsewhere](https://github.com/trishankatdatadog/hvac-openpgp/blob/246a816225cd1fbaea2baaead54d771ecf9e458b/tests/test_transit_api.py#L179-L221)